cmd/config.go

```diff
@@ -290,10 +299,16 @@ type GRPCClientConfig struct {
 	// $ dig @10.55.55.10 -t SRV _foo._tcp.service.consul +short
 	// 1 1 8080 0a585858.addr.dc1.consul.
 	// 1 1 8080 0a4d4d4d.addr.dc1.consul.
-	SRVLookup *struct {
-		Service string
-		Domain  string
-	}
+	SRVLookup *ServiceDomain
+
+	// SRVLookups allows you to pass multiple SRV records to the gRPC client.
+	// The gRPC client will resolves each SRV record and use the results to
+	// construct a list of backends to connect to. For more details, see the
+	// documentation for the SRVLookup field. Note: while you can pass multiple
+	// targets to the gRPC client using this field, all of the targets will use
+	// the same HostOverride and TLS configuration.
+	SRVLookups  []*ServiceDomain
+	srvResolver string
```

Comment:
The GRPCClientConfig struct is designed to be loaded from JSON, like several other shared configs in this file, so all its fields are public, and we can expect that a given instance of GRPCClientConfig represents something we parsed from the JSON config. We should avoid having private internal state that may (optionally) be modified after it's been loaded. That makes these objects more complicated to reason about.

Unfortunately, the way we plumb through these config values to `grpc.ClientSetup` makes it convenient to smuggle this value through in a private field rather than create a whole parallel `grpc.ClientSetupWithNonceResolver`. :-/

How about this: Make it public (`SRVResolver`) and in the WFE, require that it be set to `nonce-srv` for the GRPCClientConfig we use for the nonce service.

This has its own ugliness - we are requiring that the config have somewhat redundant information that we could figure out ourselves. But I think it's preferable to the confusion that results from making GRPCClientConfig an input / mutation / output struct as opposed to purely an input struct.

Reply:
I first took this route but the ugliness of it resulted in what it is now. I'm more than happy to revert it. I'm not a huge fan of either approach.

At this point I would almost prefer loading it from a `TXT` record within the resolver, but with (now) multiple records, that presents it's own problems unless we were to move _all_ RPC clients to use it.

Reply:
I'm now realizing this has kind of the same problem. `nonce-srv` and `srv` would both end up having their loadBalancing policy set via TXT but the default `dns` and `static` would still depend on `grpc.WithDefaultServiceConfig`.

Reply:
I'm going to ship a reversion to the version of this with the public field for `srvResolver` and moved my validation bits into an appropriate spot.